user-console/src/main/java/org/pentaho/mantle/client/ui/tabs/MantleTabPanel.java

```diff
@@ -94,6 +175,45 @@ public void addTab( String text, String tooltip, boolean closeable, Widget conte
     if ( getSelectedTab() == null ) {
       selectTab( tab );
     }
+    MantleTabMenuItem menuItem = createTabMenuItem( tab );
+    menuItemHashMap.put( tab, menuItem );
+    tabsSubMenuBar.addItem( menuItem );
+    contextMenuRefreshThreshold( true );
+    updateTabMenuText( tab );
+  }
+
+  public void renameMenuTab( PentahoTab tab ){
+    menuItemHashMap.get( tab ).setText( tab.getLabelText() );
+    updateTabMenuText( tab );
+  }
+
+  public void updateTabMenuText( PentahoTab tab ) {
+    if ( tab == null ) {
+      tabsMenuBar.addStyleName( EMPTY_TABS_MENU );
+      tabsMenuItem.setText( "" );
+    } else {
+      tabsMenuBar.removeStyleName( EMPTY_TABS_MENU );
+      String tabMenuText;
+      int tabCount = getTabCount();
+      if ( tabCount > 1 ) {
+        int tabIndex = tabsSubMenuBar.getItemIndex( menuItemHashMap.get( tab ) ) + 1;
+        tabMenuText = "Tab (" + tabIndex + "/" + tabCount + ") " + tab.getLabelText();
+      } else {
+        tabMenuText = tab.getLabelText();
+      }
+      tabsMenuItem.setText( tabMenuText );
+    }
+  }
+
+  private MantleTabMenuItem createTabMenuItem( MantleTab tab ) {
+    MantleTabMenuItem tabMenuItem = new MantleTabMenuItem( tab );
+    return tabMenuItem;
+  }
+
+  private void contextMenuRefreshThreshold( boolean added ) {
```

Comment:
The method name is not clear to me. How about `maybeRefreshContextMenus( boolean )`?

Reply:
there is a threshold between the numbers 1/2:

   Refresh the context menu when it changes from one to many.
   Disables/Enables "All Tabs" and "Other Tabs" menu items:
   - If a tab is added and there are now 2 tabs
   - If a tab is removed and there is now only 1 tab

Reply:
Thanks for the explanation. Consider adding that as a comment to the code :-)

Reply:
However, the name not being clear, or not following what I think are common expectations, still stands for me. The given name would be more appropriate for a class field or variable. Methods typically include a verb.